### PR TITLE
csvfile - let the config system do the typecasting.

### PR DESCRIPTION
**Pull request by helmo**
##### SUMMARY

In #82242 it was determined that additional cleanup was desired in existing code.
This PR resolves that.

##### ISSUE TYPE

- Bugfix Pull Request

### CODE REVIEW

**Review by bcoca — APPROVED** [review]
thanks, i was about to start doing this, but you saved me some work!

**Comment by webknjaz** [comment]
There's a traceback in the CI related to this change:
> `ansible.errors.AnsibleError: csvfile: list indices must be integers or slices, not str`

**Comment by helmo** [comment]
I don't see what's different from #82242, @bcoca hopefully you can amend this PR.

**Comment by bcoca** [comment]
#82425  should fix lookups in general

File: lib/ansible/plugins/lookup/csvfile.py [hunk 1]
```diff
@@ -135,7 +136,7 @@ def read_csv(self, filename, key, delimiter, encoding='utf-8', dflt=None, col=1,
 
             for row in creader:
                 if len(row) and row[keycol] == key:
-                    return row[int(col)]
+                    return row[col]
```
Comment (bcoca) [inline lib/ansible/plugins/lookup/csvfile.py:168]: once i merge my PR , this part won't be needed as set_option will do the casting instead

**Comment by bcoca** [comment]
/azp run

@helmo i finally got all the prereqs merged.

**Comment by helmo** [comment]
@bcoca can you try again?

**Comment by s-hertel** [comment]
The lookup_csv test is failing due to this change:
```
00:23 TASK [lookup_csvfile : Test default behavior] **********************************
00:23 task path: /root/ansible/test/results/.tmp/integration/lookup_csvfile-_sjitdkt-ÅÑŚÌβŁÈ/test/integration/targets/lookup_csvfile/tasks/main.yml:72
00:23 looking for "people.csv" at "/root/ansible/test/results/.tmp/integration/lookup_csvfile-_sjitdkt-ÅÑŚÌβŁÈ/test/integration/targets/lookup_csvfile/files/people.csv"
00:23 fatal: [testhost]: FAILED! => {
00:23     "msg": "The conditional check 'lookup('csvfile', 'notfound file=people.csv delimiter=, col=2, default=what?') == \"what?\"' failed. The error was: Invalid type for configuration option plugin_type: lookup plugin: csvfile setting: col (from Direct): invalid literal for int() with base 10: '2,'"
00:23 }
00:23 
00:23 PLAY RECAP *********************************************************************
00:23 testhost                   : ok=11   changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=4 
```

**Review by Akasurde — APPROVED** [review]

**Pull request merged**